kythe/go/platform/vfs/zip/fs.go

```diff
@@ -70,15 +70,19 @@ func (z FS) Stat(_ context.Context, path string) (os.FileInfo, error) {
 	return f.FileInfo(), nil
 }
 
-// Open implements part of vfs.Reader, returning a io.ReadCloser owned by
+// Open implements part of vfs.Reader, returning a vfs.FileReader owned by
 // the underlying zip archive. It is safe to open multiple files concurrently,
 // as documented by the zip package.
-func (z FS) Open(_ context.Context, path string) (io.ReadCloser, error) {
+func (z FS) Open(_ context.Context, path string) (vfs.FileReader, error) {
 	f := z.find(path)
 	if f == nil {
 		return nil, os.ErrNotExist
 	}
-	return f.Open()
+	fo, err := f.Open()
+	if err != nil {
+		return nil, err
+	}
+	return vfs.UnseekableFileReader{fo}, nil
```

Comment:
I *think* (Cody can correct me if I'm wrong) but the typical way this would be done is by returning a smaller interface and letting clients use typecasts to check if that type also supports a different interface?

Reply:
Either is fine.  The package already uses the `ErrNotSupported` model.